plugins/BEdita/Core/tests/TestCase/Model/Table/UserTokensTableTest.php

```diff
@@ -114,4 +115,24 @@ public function testValidFinder()
         static::assertNotEmpty($entity);
         static::assertEquals(1, $entity->get('id'));
     }
+
+    /**
+     * Test for getTokenTypes()
+     *
+     * @return void
+     * @covers ::getTokenTypes()
+     */
+    public function testGetTokenTypes()
+    {
+        $expected = UserTokensTable::DEFAULT_TOKEN_TPYES;
+        static::assertEquals($expected, $this->UserTokens->getTokenTypes());
+
+        $conf = ['token_one', 'token_two'];
+        Configure::write('UserTokens', $conf);
+        $expected = array_merge($expected, $conf);
+        static::assertEquals($expected, $this->UserTokens->getTokenTypes());
+
+        Configure::write('UserTokens', ['token_one', 'token_two', 'access']);
+        static::assertEquals($expected, $this->UserTokens->getTokenTypes());
```

Comment:
I know it's a bit overkill in this case, but if possible I'd refactor this test to use data provider.

Reply:
ahahah.... I had already refactored removing previous provider to simplify... re-re-factor is on the way :+1:

Reply:
😄 Sorry!!